File: ruby/view/textfield.rb
# Created by Satoshi Nakagawa.
# You can redistribute it and/or modify it under the Ruby's license or the GPL2.

class TextField < NSTextField
  
  def focus
    window.makeFirstResponder(self)
    e = currentEditor
    e.setSelectedRange(NSRange.new(stringValue.length, 0))
    e.scrollRangeToVisible(e.selectedRange)
  end
  
  def drawRect(rect)
    super_drawRect(rect)
    
    backgroundColor.set
    path = NSBezierPath.bezierPath
    rect = bounds
    rect.height = 3
    path.appendBezierPathWithRect(rect)
    rect = bounds
    rect.width = 2
    path.appendBezierPathWithRect(rect)
    rect = bounds
    rect.x = rect.x + rect.width - 2
    rect.width = 2
    path.appendBezierPathWithRect(rect)
    rect = bounds
    rect.y = rect.y + rect.height - 2
    rect.height = 2
    path.appendBezierPathWithRect(rect)
    path.fill
  end
  
end

File: objc/Views/TextField.m
// Created by Satoshi Nakagawa.
// You can redistribute it and/or modify it under the Ruby's license or the GPL2.

#import "TextField.h"


@implementation TextField

- (void)focus
{
	[self.window makeFirstResponder:self];
	NSText* e = [self currentEditor];
	[e setSelectedRange:NSMakeRange([[self stringValue] length], 0)];
	[e scrollRangeToVisible:[e selectedRange]];
}

@end
